types/types.go

```diff
@@ -87,32 +81,38 @@ type executionPayloadMarshallingOverrides struct {
 	LogsBloom     hexutil.Bytes
 }
 
-// SetFeeRecipientMessage as defined in https://github.com/flashbots/mev-boost/blob/main/docs/specification.md#request
-type SetFeeRecipientMessage struct {
+// RegisterValidatorRequestMessage as defined in https://github.com/flashbots/mev-boost/blob/main/docs/specification.md#request
+type RegisterValidatorRequestMessage struct {
 	FeeRecipient string `json:"feeRecipient"`
 	Timestamp    string `json:"timestamp"`
+	Pubkey       string `json:"pubkey"`
+}
+
+// IsValid returns true if the message properties are correct valid hex values
+func (r *RegisterValidatorRequestMessage) IsValid() bool {
```

Comment:
I think this check is not necessary, as it's not really a validity check
I'd leave it non-validated until we have proper validation (parse `FeeRecipient` to `common.Address`, `Timestamp` to uint64, `Pubkey` to a proper pubkey
I think I'd also apply this to the rest of the validation - partial validation (in this case - making sure it's a hex) seems unnecessary

Reply:
I will see if I can add types that better catch issues, but generally I think it's better to reject obviously wrong input, even if it's not perfect, rather than not checking at all.

Reply:
Updated with proper types